java/src/org/exist/jms/replication/subscribe/ReplicationJmsListener.java

```diff
@@ -729,9 +720,9 @@ private Collection getOrCreateCollection(final XmldbURI sourcePath, final Option
             throw new MessageReceiveException(t.getMessage(), t);
 
         } finally {
-            releaseLock(newCollection, Lock.LockMode.WRITE_LOCK);
+            releaseLock(collection, Lock.LockMode.WRITE_LOCK);
```

Comment:
I have been struggling with this subject for many years, it got never clear to me how things worked. Lack of documentation :-)